openmdao/core/tests/test_connections.py

```diff
@@ -613,5 +613,66 @@ def compute_partials(self, inputs, J):
         else:
             self.fail('Exception expected.')
 
+@unittest.skipUnless(MPI, "MPI is required.")
+class TestConnectionsError(unittest.TestCase):
+    N_PROCS = 2
+
+    def test_incompatible_src_indices(self):
+        SRC_INDICES = [1, 2]
+        FORCE_PETSCTRANSFER = True
+        COMP_DISTRIBUTED = False
+
+        class TestCompDist(om.ExplicitComponent):
+        # this comp is distributed and forces PETScTransfer
+            def initialize(self):
+                self.options['distributed'] = True
+
+            def setup(self):
+                self.add_input('x', shape=2)
+                self.add_output('y', shape=1)
+                self.declare_partials('y', 'x', val=1.0)
+
+            def compute(self, inputs, outputs):
+                outputs['y'] = np.sum(inputs['x'])
+
+        class TestComp(om.ExplicitComponent):
+            def initialize(self):
+                self.options['distributed'] = COMP_DISTRIBUTED
+
+            def setup(self):
+                # read SRC_INDICES on each proc
+                self.add_input('x', shape=2, src_indices=SRC_INDICES, val=-2038.0)
+                self.add_output('y', shape=1)
+                self.declare_partials('y', 'x')
+
+            def compute(self, inputs, outputs):
+                outputs['y'] = np.sum(inputs['x'])
+
+            def compute_partials(self, inputs, J):
+                J['y', 'x'] = np.ones((2,))
+
+        prob = om.Problem()
+        model = prob.model
+
+        if MPI.COMM_WORLD.rank == 0:
+            setval = np.array([2.0,3.0])
+        else:
+            setval = np.array([10.0,20.0])
+
+        # no parallel or distributed comps, so default_vector is used (local xfer only)
+        model.add_subsystem('p1', om.IndepVarComp('x', setval))
+        model.add_subsystem('c3', TestComp())
+        if FORCE_PETSCTRANSFER:
+            model.add_subsystem('c4', TestCompDist())
```

Comment:
I think the intent was to add a test that runs 
```under MPI (without adding any distributed comps or parallelgroups to the model) and see if a correct error is raised.```
(quoting @Kenneth-T-Moore from #1467 discussion)
So the `FORCE_PETSCTRANSFER=False` version of @bbrelje 's gist?

Reply:
hmm I see the `COMP_DISTRIBUTED` flag does this.. so not clear on the function of the FORCE_PETSCTRANSFER flag

Reply:
Yeah I could clean that up a bit. I'll fix it